source/adapters/level_zero/adapter.cpp

```diff
@@ -35,19 +35,41 @@ class ur_legacy_sink : public logger::Sink {
   ~ur_legacy_sink() = default;
 };
 
-ur_result_t initPlatforms(PlatformVec &platforms) noexcept try {
+ur_result_t initPlatforms(PlatformVec &platforms,
+                          ze_result_t ZesResult) noexcept try {
   uint32_t ZeDriverCount = 0;
   ZE2UR_CALL(zeDriverGet, (&ZeDriverCount, nullptr));
   if (ZeDriverCount == 0) {
     return UR_RESULT_SUCCESS;
   }
 
+  uint32_t ZesDriverCount = 0;
+  ze_result_t zesDriverGetResult = ZE_RESULT_ERROR_UNINITIALIZED;
+  if (ZesResult == ZE_RESULT_SUCCESS) {
+    zesDriverGetResult =
+        ZE_CALL_NOCHECK(zesDriverGet, (&ZesDriverCount, nullptr));
+  }
+
   std::vector<ze_driver_handle_t> ZeDrivers;
   ZeDrivers.resize(ZeDriverCount);
 
+  std::vector<zes_driver_handle_t> ZesDrivers;
+  ZesDrivers.resize(ZesDriverCount);
+
   ZE2UR_CALL(zeDriverGet, (&ZeDriverCount, ZeDrivers.data()));
+  if (zesDriverGetResult == ZE_RESULT_SUCCESS && ZesDriverCount > 0) {
+    ZE_CALL_NOCHECK(zesDriverGet, (&ZesDriverCount, ZesDrivers.data()));
+  }
   for (uint32_t I = 0; I < ZeDriverCount; ++I) {
+    zes_driver_handle_t zesDriverHandle = nullptr;
+    // If the number of ZE Drivers is greater than the number of ZES Drivers,
+    // then a null sysman driver is associated. The index of the ZE and ZES
+    // driver should match.
+    if (I < ZesDrivers.size()) {
+      zesDriverHandle = ZesDrivers[I];
```

Comment:
hmm how can we be sure the drivers are enumerated in same order? but having said that i see core driver_properties at least provide a a driver_uuid but Sysman does not have that.